src/Compilers/CSharp/Portable/Symbols/Source/ExplicitInterfaceHelpers.cs

```diff
@@ -176,12 +176,17 @@ public static string GetMemberNameWithoutInterfaceName(string fullName)
 
             var memberLocation = implementingMember.Locations[0];
             var containingType = implementingMember.ContainingType;
-            var containingTypeKind = containingType.TypeKind;
 
-            if (containingTypeKind != TypeKind.Class && containingTypeKind != TypeKind.Struct)
+            switch (containingType.TypeKind)
             {
-                diagnostics.Add(ErrorCode.ERR_ExplicitInterfaceImplementationInNonClassOrStruct, memberLocation, implementingMember);
-                return null;
+                case TypeKind.Class:
+                case TypeKind.Struct:
+                case TypeKind.Interface:
+                    break;
+
+                default:
+                    diagnostics.Add(ErrorCode.ERR_ExplicitInterfaceImplementationInNonClassOrStruct, memberLocation, implementingMember);
```

Comment:
> Do we have a test that exercises this line?

Yes, ExplicitImplementation in src\compilers\csharp\test\emit\codegen\codegenscripttests.cs.